dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java

```diff
@@ -0,0 +1,188 @@
+
+    private final static int THREAD_AMOUNT = 7;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * The main purpose of this test is to verify the logic applied by the {@link DotJobStore} class
+     * to create triggers for Stateful jobs
+     * <b>Method to test:</b> {@link DotJobStore#storeJobAndTrigger(SchedulingContext, JobDetail,
+     * Trigger)}
+     * <p>
+     * <b>Given scenario:</b> Multiple threads trigger jobs in a stateful job
+     * <p>
+     * <b>Expected result:</b> Only one trigger should be created to handle all the jobs
+     */
+    @Test
+    public void testDotStatefulJobIsReallyStateful()
+            throws SchedulerException, InterruptedException {
+        Scheduler scheduler = QuartzUtils.getSequentialScheduler();
+
+        final CyclicBarrier cyclicBarrier = new CyclicBarrier(THREAD_AMOUNT);
+        final CountDownLatch countDownLatch = new CountDownLatch(THREAD_AMOUNT);
+
+        for (int i = 0; i < THREAD_AMOUNT; i++) {
+            final Thread worker = new Thread(
+                    new StatefulTestThread(cyclicBarrier, countDownLatch, scheduler));
+            worker.setName("Thread " + i);
+            worker.start();
+        }
+
+        //wait until all threads are executed to verify the assert
+        countDownLatch.await(15, TimeUnit.SECONDS);
+
+        assertEquals(1, scheduler.getTriggerNames("StatefulTestJob_trigger_group").length);
+    }
+
+
+    /**
+     * <b>Method to test:</b> {@link TriggerBuilder#build()}
+     * <p>
+     * <b>Given scenario:</b> Multiple threads trigger jobs in a stateful job
+     * <p>
+     * <b>Expected result:</b> Only one trigger should be created to handle all the jobs
+     */
+    @Test
+    public void testTriggerBuilder() throws InterruptedException {
+
+        final CyclicBarrier cyclicBarrier = new CyclicBarrier(THREAD_AMOUNT);
+        final CountDownLatch countDownLatch = new CountDownLatch(THREAD_AMOUNT);
+        final DotStatefulTestJob dotStatefulTestJob = new DotStatefulTestJob();
+
+        for (int i = 0; i < THREAD_AMOUNT; i++) {
+            final Thread worker = new Thread(
+                    new StatefulBuilderTestThread(cyclicBarrier, countDownLatch, dotStatefulTestJob));
+            worker.setName("Thread " + i);
+            worker.start();
+        }
+
+        //wait until all threads are executed to verify the assert
+        countDownLatch.await(15, TimeUnit.SECONDS);
+
+        assertTrue(DotStatefulTestJob.getThreadsExecuted() > 1);
+    }
+
+    /**
+     * Inner class with the logic to be executed by each thread in {@link
+     * DotStatefulJobTest#testDotStatefulJobIsReallyStateful()} ()}
+     */
+    private class StatefulTestThread implements Runnable {
+
+        private CyclicBarrier cyclicBarrier;
+        private CountDownLatch countDownLatch;
```

Comment:
Codacy found an issue: [Private field 'countDownLatch' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5971105)